File: go/elons-toys/elons_toys.go
package elon

import "fmt"

// TODO: define the 'Drive()' method
func (c *Car) Drive() {
	if c.battery >= c.batteryDrain {
		c.battery -= c.batteryDrain
		c.distance += c.speed
	}
}

// TODO: define the 'DisplayDistance() string' method
func (c *Car) DisplayDistance() string {
	return fmt.Sprintf("Driven %d metres", c.distance)
}

// TODO: define the 'DisplayBattery() string' method

// TODO: define the 'CanFinish(trackDistance int) bool' method

File: csharp/elons-toys/ElonsToys.cs
using System;

class RemoteControlCar
{
    public int Distance = 0;
    public int BatteryPercentage = 100;
    public static RemoteControlCar Buy()
    {
        RemoteControlCar newCar = new();
        return newCar;
    }

    public string DistanceDisplay()
    {
        return $"Driven {Distance} meters";
    }

    public string BatteryDisplay()
    {
        if (BatteryPercentage < 1)
        {
            return "Battery empty";
        }
        return $"Battery at {BatteryPercentage}%";
    }

    public void Drive()
    {
        if (BatteryPercentage > 0)
        {
            Distance += 20;
            BatteryPercentage -= 1;
        }

    }
}
